### ISSUE TITLE
net/http error can result in a nil *http.Response which causes panic

### ISSUE THREAD

**Issue by bkeroackdsc** [google/go-github#659]
I was investigating this crash:

```
furan_1      | panic: runtime error: invalid memory address or nil pointer dereference
furan_1      | [signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x7bb77a]
furan_1      |
furan_1      | goroutine 911 [running]:
furan_1      | github.com/dollarshaveclub/furan/vendor/github.com/google/go-github/github.(*Response).populatePageValues(0xc42040d180)
furan_1      | 	/go/src/github.com/dollarshaveclub/furan/vendor/github.com/google/go-github/github/github.go:294 +0x3a
furan_1      | github.com/dollarshaveclub/furan/vendor/github.com/google/go-github/github.newResponse(0x0, 0xc420435c00)
furan_1      | 	/go/src/github.com/dollarshaveclub/furan/vendor/github.com/google/go-github/github/github.go:286 +0x5a
furan_1      | github.com/dollarshaveclub/furan/vendor/github.com/google/go-github/github.(*RepositoriesService).GetArchiveLink(0xc420335b90, 0xc420363c00, 0xf, 0xc420363c10, 0x4, 0xc77aef, 0x7, 0xc42004bb68, 0x1d40e0e9f774ab6f, 0xa03bb7934bbb0dbe, ...)
furan_1      | 	/go/src/github.com/dollarshaveclub/furan/vendor/github.com/google/go-github/github/repos_contents.go:271 +0x273
furan_1      | github.com/dollarshaveclub/furan/lib.(*GitHubFetcher).Get(0xc42000ec10, 0xc420363c00, 0xf, 0xc420363c10, 0x4, 0xc42020768a, 0x6, 0x1, 0x106bb30, 0xc4204b4000, ...)
furan_1      | 	/go/src/github.com/dollarshaveclub/furan/lib/github_fetch.go:58 +0xc9
furan_1      | github.com/dollarshaveclub/furan/lib.(*ImageBuilder).Build(0xc4204a71e0, 0x7f03ad986eb0, 0xc42032ae00, 0xc420207550, 0x1d40e0e9f774ab6f, 0xa03bb7934bbb0dbe, 0x0, 0x1d40e0e9f774ab6f, 0xa03bb7934bbb0dbe, 0x2)
furan_1      | 	/go/src/github.com/dollarshaveclub/furan/lib/builder.go:229 +0x420
furan_1      | github.com/dollarshaveclub/furan/lib.(*GrpcServer).syncBuild(0xc4200c9bc0, 0x7f03ad986eb0, 0xc42032ae00, 0xc420207550, 0xc400000000)
furan_1      | 	/go/src/github.com/dollarshaveclub/furan/lib/grpc.go:313 +0x3f4
furan_1      | github.com/dollarshaveclub/furan/lib.(*GrpcServer).buildWorker(0xc4200c9bc0, 0x7f03ad986eb0, 0xc4202c3600)
furan_1      | 	/go/src/github.com/dollarshaveclub/furan/lib/grpc.go:166 +0x2bd
furan_1      | github.com/dollarshaveclub/furan/lib.(*GrpcServer).runWorkers.func1(0xc4200c9bc0, 0x7f03ad986eb0, 0xc4202c3600)
furan_1      | 	/go/src/github.com/dollarshaveclub/furan/lib/grpc.go:153 +0x6c
furan_1      | created by github.com/dollarshaveclub/furan/lib.(*GrpcServer).runWorkers
furan_1      | 	/go/src/github.com/dollarshaveclub/furan/lib/grpc.go:153 +0x14d
```

Which is ultimately caused by an erroneous newline in the oauth2 token, which causes `net/http` to throw this error: `net/http: invalid header field value "Bearer xxxxxxxxxxxxxxxxxxxx\n" for key Authorization`

TLDR: ``newResponse()`` should check for a nil *http.Response before calling ``response.populatePageValues()``

**Comment by shurcooL** [comment]
```
github.com/dollarshaveclub/furan/vendor/github.com/google/go-github/github.(*RepositoriesService).GetArchiveLink(0xc420335b90, 0xc420363c00, 0xf, 0xc420363c10, 0x4, 0xc77aef, 0x7, 0xc42004bb68, 0x1d40e0e9f774ab6f, 0xa03bb7934bbb0dbe, ...)
/go/src/github.com/dollarshaveclub/furan/vendor/github.com/google/go-github/github/repos_contents.go:271 +0x273
```

What version of `go-github` were you using? I don't see a line 271 in [`repos_contents.go`](https://github.com/google/go-github/blob/7a51fb928f52a196d5f31daefb8a489453ef54ff/github/repos_contents.go#L266) on `master`.

Okay, your repo is public, and I can see myself:

https://github.com/dollarshaveclub/furan/blob/441118307f95d9bd4750f74262a1a622bcc42257/vendor/github.com/google/go-github/github/repos_contents.go#L271

I think we've already fixed that bug since then. Look at #485. `newResponse` is no longer called when `resp` is `nil`.
